xapi/tipb/select.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto2";
+
+package tipb;
+
+option java_multiple_files = true;
+option java_package = "com.pingcap.tidb.tipb";
+option java_outer_classname = "SelectProto";
+
+import "expression.proto";
+import "schema.proto";
+
+// KeyRange is the encoded index key ranges, low is closed, high is open. (low <= x < high)
+message KeyRange {
+	optional bytes low = 1;
+	optional bytes high = 2;
+}
+
+// ByItem type for group by and order by.
+message ByItem {
+	optional Expr expr = 1;
+	optional bool desc = 2;
+}
+
+// SelectRequest works like a simplified select statement.
+message SelectRequest {
+	// transaction start timestamp.
+	optional int64 start_ts = 1;
+
+	// If table_info is not null, it represents a table scan, index_info would be null.
+	optional TableInfo table_info = 2;
+
+	// If IndexInfo is not null, it represents a index scan, table_info would be null.
+	optional IndexInfo index_info = 3;
+
+	// fields to be selected, fields type can be column reference for simple scan.
+	// or aggregation function. If no fields specified, only handle will be returned.
```

Comment:
How about "select substring(c, 1) from t"? Can we push substring down to coprocessor?